flink/src/main/scala/org/apache/zeppelin/flink/FlinkScalaInterpreter.scala

```diff
@@ -231,3 +233,71 @@ class FlinkScalaInterpreter(val properties: Properties) {
   def getBatchTableEnviroment(): BatchTableEnvironment = this.btenv
 
 }
+
+private object FlinkScalaInterpreter {
+
+  /**
+    * This is a hack to call `loopPostInit` at `ILoop`. At higher version of Scala such
+    * as 2.11.12, `loopPostInit` became a nested function which is inaccessible. Here,
+    * we redefine `loopPostInit` at Scala's 2.11.8 side and ignore `loadInitFiles` being called at
+    * Scala 2.11.12 since here we do not have to load files.
+    *
+    * Both methods `loopPostInit` and `unleashAndSetPhase` are redefined, and `phaseCommand` and
+    * `asyncMessage` are being called via reflection since both exist in Scala 2.11.8 and 2.11.12.
+    *
+    * Please see the codes below:
+    * https://github.com/scala/scala/blob/v2.11.8/src/repl/scala/tools/nsc/interpreter/ILoop.scala
+    * https://github.com/scala/scala/blob/v2.11.12/src/repl/scala/tools/nsc/interpreter/ILoop.scala
+    *
+    * See also ZEPPELIN-3810/ZEPPELIN-4187.
+    */
+  private def loopPostInit(interpreter: FlinkScalaInterpreter): Unit = {
+    import StdReplTags._
+    import scala.reflect.{classTag, io}
+
+    val sparkILoop = interpreter.flinkILoop
```

Comment:
`sparkILoop`?

Reply:
`callMethod(flinkILoop, "scala$tools$nsc$interpreter$ILoop$$loopPostInit")` was failing so got it replaced with the help of ZEPPELIN-3810, let me fix the variable name.